src/users/users.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, ParseIntPipe } from '@nestjs/common';
+import { UsersService } from './users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('users')
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Post()
+  @ApiOperation({summary: 'Create new User'})
+  @ApiResponse({ status: 201, description: 'User created successfully' })
+  @ApiResponse({ status: 400, description: 'Bad Request. Missing or invalid inputs.' })
+  @ApiResponse({ status: 500, description: 'Internal Server Error' })
+  create(@Body() createUserDto: CreateUserDto) {
+    return this.usersService.create(createUserDto);
+  }
+
+  @Get()
+  @ApiOperation({ summary: 'Returns all registered User' })
+  @ApiResponse({ status: 200, description: 'Returns an array of users.' })
+  findAll() {
+    return this.usersService.findAll();
```

Comment:
this should be only accessible if you are logged and shouldn't return the users passwords